### PR TITLE
Add machines.yaml sample

### PR DESCRIPTION
**Pull request by wangtjia**
Add `machines.yaml` sample

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Specify your PR reviewers**:

/cc @gyliu513 @jichenjc @xunpan

**Special notes for your reviewer**:

/sig ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,6 +134,8 @@ kube-system                kubernetes-dashboard-5f7b999d65-ntrxb         1/1
 2. Create a cluster:
 
 You should review files `cluster.yaml` and `machines.yaml`, and ensure the `domain`, `dataCenter` and `osReferenceCode` are set. You can customize those values based on your requirement.
+For `machines.yaml`, you can follow the [sample file](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/tree/master/examples/ubuntu), and make sure `sshKeyName` and the label in `ibmcloud` are identical.
```
Comment (gyliu513) [inline README.md]: Please use link https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/master/examples/ubuntu/machines.yaml directly.

Comment (gyliu513) [inline README.md]: Add a blank line before Line 137

**Comment by wangtjia** [comment]
I signed it

**Comment by wangtjia** [comment]
I signed it

File: README.md [hunk 2]
```diff
@@ -135,6 +135,9 @@ kube-system                kubernetes-dashboard-5f7b999d65-ntrxb         1/1
 
 You should review files `cluster.yaml` and `machines.yaml`, and ensure the `domain`, `dataCenter` and `osReferenceCode` are set. You can customize those values based on your requirement.
 
+For `machines.yaml`, you can follow the [sample file](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/master/examples/ubuntu/machines.yaml), and make sure `sshKeyName` and the label in `ibmcloud` are identical.
+
```
Comment (gyliu513) [inline README.md]: kill this line.

Also please ask @songleo for how to squash your commit to one commit.

Comment (jichenjc) [reply README.md]: I don't fully understand 
`sshKeyName` and the label in `ibmcloud` are identical.

can you help to provide detail?

Comment (wangtjia) [reply README.md]: The `sshKeyName` in `machines.yaml` should be identical to the `label` value in Softlayer SSH KEY

image

image

Comment (jichenjc) [reply README.md]: you may add those pictures, for reader to understand what's the meaning here
and please note actually you can have more ssh keys created in cloud and you will select one of them in the machines.yaml

Comment (wangtjia) [reply README.md]: I will add those pictures. Thanks for your advice.

**Pull request merged**